### DISCUSSION TITLE
How to set  specific seed coordinates at both scales（ I have discovered a bug ）

### DISCUSSION THREAD

**Discussion by siwux** [neperfepx/neper#411] [Q&A] [answered]
I want to generate a  2D model with multi-scale tessellation. For example：
![image](https://user-images.githubusercontent.com/90365358/159111963-26128c18-49e7-4103-bda5-06fd88023780.png)
`neper -T -dim 2 -domain "square(5,4)" 
-n "2::msfile(file1)::msfile(file2)::msfile(file3)" 
-morphooptiini "coo:file(coo1),weight:0::coo:file(coo2),weight:0" 
-o simulation`
file1:
1 1
2 2
file2:
1::1 1
2::1 2
2::2 2
file3:
1::1::1 1
2::1::1 1
2::1::2 2
2::2::1 1
2::2::2 2
coo1:
1 2
3 2
coo2:
1 1 2
2 3 1
2 3 3
I don't know how to set the seed coordinates of 3th scale,so I can just get the model:
![image](https://user-images.githubusercontent.com/90365358/159112405-8fdf6c74-092d-411c-8419-97c2a33aacfc.png)
I get wrong with this cide:
`neper -T -dim 2 -domain "square(5,4)" 
-n "2::msfile(file1)::msfile(file2)::msfile(file3)" 
-morphooptiini "coo:file(coo1),weight:0::coo:file(coo2),weight:0::coo:file(coo3),weight:0::coo:file(coo4),weight:0" 
-o test`
file1:
1 1
2 2
file2:
1::1 1
2::1 2
2::2 2
file3:
1::1::1 1
2::1::1 1
2::1::2 2
2::2::1 1
2::2::2 2
coo1:
1 2
3 2
coo2:
1 1 2
2 3 1
2 3 3
coo3:
1::1 1 2
2::1 3 1
2::1 4.5 1
2::2 3 3
2::2 4.5 3
coo4:
1::1::1 1 2
2::1::1 3 1
2::1::2 4.5 0.5
2::1::2 4.5 1.5
2::2::1 3 3
2::2::2 4.5 2.5
2::2::2 4.5 3.5
![image](https://user-images.githubusercontent.com/90365358/159113304-93035146-1819-4b94-a7c1-d763fa368423.png)

**Comment by rquey** [comment, marked as answer]
The structure can be seen as made of lamellae, so, I would go:

```
neper -T -dim 2 -n from_morpho::from_morpho -morpho "lamellar(w=0.4,v=(1,0,0),pos=start)::msfile(n_ms_file)" -reg 1 -o d
```

where morpho_msfile is:
```
1 lamellar(w=1.00,v=(0,1,0),pos=start)
2 lamellar(w=0.50,v=(0,1,0),pos=start)
3 lamellar(w=0.249999,v=(0,1,0),pos=start)
```

The reason why we must use `0.249999` instead of `0.25` is that `0.25` would create a degenerate configuration, which is to avoid. This is "fixed" by `-reg 1`, which removes the very small edge that results from the use of `0.249999` (at the intersection between the blue, green, cyan and magenta cells).

Visualize:
![image](https://user-images.githubusercontent.com/2659376/159435748-ecee15df-41ee-4596-ba8b-b36c1072c024.png)

**Reply by siwux** [reply]
Thank you very much for your reply.
